### ISSUE TITLE
Deadlock when using shardByURL of VMAgent

### ISSUE THREAD

**Issue by penguinlav** [VictoriaMetrics/VictoriaMetrics#5833]
### Describe the bug

When I send metrics to vmagent with shardByURL configured,  VMAgent goes to deadlock.

### To Reproduce

Start VMAgent
```
vmagent --remoteWrite.url=http://host01,http://host02 --graphiteListenAddr=:2003 --remoteWrite.shardByURL=true
```

send metrics
```
echo "test 100 `date +%s`" | nc localhost -c 2003
```

`nc` connection does not close and vmagent does not respond to signals

### Version

1.97.1, 1.98.0

**Comment by valyala** [comment]
FYI, this bug has been fixed in [v1.93.13 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.93.13). The bugfix will be included in the latest release too.

**Comment by valyala** [comment]
FYI, this bug has been fixed also in [v1.97.3 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.97.3).

**Comment by valyala** [comment]
FYI, this bug has been fixed starting from [v1.99.0 release](https://docs.victoriametrics.com/changelog/). Closing the issue as fixed.
